src/c++/perf_analyzer/mock_data_loader.h

```diff
@@ -44,6 +44,9 @@ class MockDataLoader : public DataLoader {
 
     return ParseData(d, inputs, outputs);
   };
+
+  std::vector<size_t>& step_num_{DataLoader::step_num_};
```

Comment:
I need to modify this aspect of the mock's behavior to make my test work properly. Any other suggestions on how to accomplish this?

Reply:
If it is really a mock then you should be able to mock out the function instead of the member variable. 

Seems like these are the functions that matter?
  size_t GetDataStreamsCount()
  size_t GetTotalSteps(size_t stream_id)

Could override them to return a fixed value that you can configure.

Reply:
> to return a fixed value that you can configure.

This is me configuring a fixed value for those functions to return.

Reply:
I really don't like this. It is a weird hybrid of mock and real class. It is making assumptions about how the private values are used by the real class, and thus the testing of sequence_manager is coupled with the implementation details of data_loader.  I think the proper way is to define the interface, and then have the mock classes implement those functions. Regardless, we know we have a big follow-on story to standardize our testing, so I'll just defer this until then.